File: src/main/java/kitchenpos/products/tobe/domain/ProductNamePolicy.java
package kitchenpos.products.tobe.domain;

import kitchenpos.common.DomainService;

import java.util.Objects;

@DomainService
public class ProductNamePolicy {
    private final ProductNameProfanities productNameProfanities;

    public ProductNamePolicy(ProductNameProfanities productNameProfanities) {
        this.productNameProfanities = productNameProfanities;
    }

    public void validateDisplayName(Name displayName) {
        if (Objects.isNull(displayName.getValue()) || this.containsProfanity(displayName)) {
            throw new IllegalArgumentException();
        }
    }

    private boolean containsProfanity(Name name) {
        return productNameProfanities.containsProfanity(name.getValue());
    }
}

File: src/main/kotlin/kitchenpos/product/tobe/domain/ProductNamePolicy.kt
package kitchenpos.product.tobe.domain

import kitchenpos.common.annotation.DomainService
import kitchenpos.common.domain.Profanities

@DomainService
class ProductNamePolicy(private val profanities: Profanities) {
    fun validate(name: String) {
        require(name.isNotBlank()) { "상품 이름은 필수값입니다." }
        require(!profanities.contains(name)) { "상품 이름에 금지어가 포함되어 있습니다." }
    }
}
